torchao/quantization/quantize_/workflows/int4/int4_tensor.py

```diff
@@ -43,23 +44,25 @@ class Int4Tensor(TorchAOBaseTensor):
     """
 
     tensor_data_names = ["qdata", "scale", "zero_point"]
+    optional_tensor_data_names = ["activation_scale"]
     tensor_attribute_names = ["block_size", "shape"]
 
-    def __new__(cls, qdata, scale, zero_point, block_size, shape):
+    def __new__(cls, qdata, scale, zero_point, activation_scale, block_size, shape):
```

Comment:
I feel like an optional argument like this should be last, after non-optional ones like block_size and shape

Reply:
got it

is this kind of change bc breaking? if yes, is there an opportunity to make all of these keyword-only arguments to remove from bc?

Reply:
if we have a stable release of these, then yes, although right now we don't have these in any of the official stable release yet

some of the data args are required for the construction to make sense I think, do you mean making all attributes keyword-only arg?